services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java

```diff
@@ -66,27 +66,33 @@
 @SdkInternalApi
 public final class DefaultS3TransferManager implements S3TransferManager {
     private static final Logger log = Logger.loggerFor(S3TransferManager.class);
-    private final S3CrtAsyncClient s3CrtAsyncClient;
+    private final S3AsyncClient s3AsyncClient;
     private final TransferManagerConfiguration transferConfiguration;
     private final UploadDirectoryHelper uploadDirectoryHelper;
     private final DownloadDirectoryHelper downloadDirectoryHelper;
 
     public DefaultS3TransferManager(DefaultBuilder tmBuilder) {
         transferConfiguration = resolveTransferManagerConfiguration(tmBuilder);
-        s3CrtAsyncClient = initializeS3CrtClient(tmBuilder);
+        s3AsyncClient = tmBuilder.s3AsyncClient;
         uploadDirectoryHelper = new UploadDirectoryHelper(transferConfiguration, this::uploadFile);
-        ListObjectsHelper listObjectsHelper = new ListObjectsHelper(s3CrtAsyncClient::listObjectsV2);
+        ListObjectsHelper listObjectsHelper = new ListObjectsHelper(s3AsyncClient::listObjectsV2);
         downloadDirectoryHelper = new DownloadDirectoryHelper(transferConfiguration,
                                                               listObjectsHelper,
                                                               this::downloadFile);
+
+        if (!(s3AsyncClient instanceof S3CrtAsyncClient)) {
+            log.warn(() -> "s3AsyncClient must be of S3CrtAsyncClient in order to get maximum throughput. The current"
+                           + " s3AsyncClient is of " +s3AsyncClient.getClass().getSimpleName() +" type in which individual file"
```

Comment:
Maybe something like the following?
```
The provided S3AsyncClient is not an instance of S3CrtAsyncClient, and thus multipart 
upload/download feature is not enabled. To benefit from maximum throughput, 
consider using S3AsyncClient.crtBuilder().build() instead.
```
On a side note, getSimpleName has poor performance.

Reply:
Thanks